webapi/Services/SemanticKernelProvider.cs

```diff
@@ -159,7 +162,9 @@ IMemoryStore CreateMemoryStore()
             switch (memoryOptions.Retrieval.VectorDbType)
             {
                 case string x when x.Equals("SimpleVectorDb", StringComparison.OrdinalIgnoreCase):
-                    return new VolatileMemoryStore();
+                    // Maintain single instance of volitile memory.
+                    Interlocked.CompareExchange(ref _memoryStore, new VolatileMemoryStore(), null);
```

Comment:
I notice you are a big fan of CompareExchange.  That's all good.  But for creating singletons, I like using Lazy<t>. If you don't know about Lazy, check it out. I find it cleaner to create singletons.

This is just a public information announcement,  Feel free to ignore,

Reply:
Lazy had some well known problems...not sure if those got addressed in .net core, but there's a historical reason people avoid Lazy.

Reply:
Yeah, just did double-check...not a fan of `Lazy`.  I think the highest bar here would be to define VolatileMemoryStore as a singleton and then sp.GetRequiredService()...but because of the DI layering, its a bit of a pita here (and just not worth it).

FWIW - This code will be removed after a few more releases.